source/configuration_manual/authentication/oauth2.rst

```diff
@@ -98,41 +113,32 @@ passdb settings
 
 .. code-block:: none
 
-  auth_oauth2_config_file = /usr/local/etc/dovecot/dovecot-oauth2.token.conf.ext
-
   passdb oauth2 {
-    mechanisms = plain login
-    args = /usr/local/etc/dovecot/dovecot-oauth2.plain.conf.ext
+    oauth2 {
+      # common settings can be inherited from global scope
+      mechanisms = plain login
+      grant_url = http://localhost:8000/token
+      use_grant_password = yes
+      fields {
+         host = 127.0.0.1
+         proxy = y
+         proxy_mech = xoauth2
+         pass = %{oauth2:access_token}
+      }
   }
 
-put into ``dovecot-oauth2.token.conf.ext``
-
-.. code-block:: none
-
-  grant_url = http://localhost:8000/token
-  client_id = verySecretClientId
-  client_secret = verySecretSecret
-  tokeninfo_url = http://localhost:8000/oauth2?oauth=
-  introspection_url = http://localhost:8000/introspect
-  introspection_mode = post
-  use_grant_password = no
-  debug = yes
-  username_attribute = username
-  pass_attrs = pass=%{oauth2:access_token}
-
-put into ``dovecot-oauth2.plain.conf.ext``
-
-.. code-block:: none
-
-  grant_url = http://localhost:8000/token
-  client_id = verySecretClientId
-  client_secret = verySecretSecret
-  introspection_url = http://localhost:8000/introspect
-  introspection_mode = post
-  use_grant_password = yes
-  debug = yes
-  username_attribute = username
-  pass_attrs = host=127.0.0.1 proxy=y proxy_mech=xoauth2 pass=%{oauth2:access_token}
+  oauth2 {
+    client_id = verySecretClientId
+    client_secret = verySecretSecret
+    tokeninfo_url = http://localhost:8000/oauth2?oauth=
+    introspection_url = http://localhost:8000/introspect
+    introspection_mode = post
+    use_grant_password = no
+    username_attribute = username
+    fields {
+       pass = %{token}
```

Comment:
`%{oauth2:access_token}` like elsewhere?

Reply:
in fact you can use %{token}, as db-oauth2 adds that to the passdb.

Reply:
It should then be `%{passdb:token}`